File: backend/app/schemas/workflow.py
"""Pydantic schemas for workflow control — HitlFeedback + responses — §8.3.2."""

from pydantic import BaseModel, Field


class HitlFeedback(BaseModel):
    """User feedback submitted at an HITL interrupt node."""

    hitl_type: str = Field(
        ...,
        pattern=r"^(search_review|outline_review|draft_review)$",
        description="Type of HITL node",
    )

    # ── Search review ──
    selected_paper_ids: list[str] | None = None
    additional_query: str | None = None

    # ── Outline review ──
    approved_outline: dict | None = None

    # ── Draft review ──
    revision_instructions: str | None = None

    approved: bool = True


class WorkflowStartResponse(BaseModel):
    """Response after starting/resuming a workflow."""

    task_id: str
    status: str


class WorkflowStatusResponse(BaseModel):
    """Current workflow status."""

    project_id: str
    phase: str | None
    status: str
    progress: dict | None = None
    token_usage: dict | None = None


class ExportRequest(BaseModel):
    """Export format request."""

    format: str = Field(
        ...,
        pattern=r"^(markdown|word|bibtex|ris)$",
        description="Export format",
    )

File: frontend/src/types/workflow.ts
/** HITL 反馈请求体 — 对齐后端 HitlFeedback */
export interface HitlFeedback {
    hitl_type: 'search_review' | 'outline_review' | 'draft_review';
    selected_paper_ids?: string[];
    additional_query?: string;
    approved_outline?: Record<string, unknown>;
    revision_instructions?: string;
    approved?: boolean;
}

/** 工作流启动响应 — 对齐后端 WorkflowStartResponse */
export interface WorkflowStartResponse {
    task_id: string;
    status: string;
}

/** 工作流状态响应 — 对齐后端 WorkflowStatusResponse */
export interface WorkflowStatusResponse {
    project_id: string;
    phase: string | null;
    status: string;
    progress: Record<string, unknown> | null;
    token_usage: Record<string, number> | null;
}

/** 导出请求体 — 对齐后端 ExportRequest */
export interface ExportRequest {
    format: 'markdown' | 'word' | 'bibtex' | 'ris';
}
